### PR TITLE
Create raw_metrics_op for creating PR curves

### LINKED ISSUE #515: Add a custom summary op for PR curves that takes raw precision, recall, TP, FP, TN, FN.
An internal team has a custom way of computing PR curves. For their use case, true positives are rare, so  the current PR curves plots produced by TensorBoard are noisy for them.

To solve this problem, they made a streaming metric that accumulates weighted (TP, FP, TN, FN) counts across a certain window of recent steps, which has a smoothing effect and makes the PR curves jump less from step to step.

To accommodate that, I am thinking about adding a raw_precision_recall_op method to the summary.py file for PR curves. I started this issue to get everyone's input on that.

@wchargin @jart @dandelionmane

### PR DESCRIPTION
**Pull request by chihuahua**
This change introduces a `raw_metrics_op` for collecting data for generating PR curves. Fixes #515. See #515 for the motivation behind `raw_metrics_op`.

### CODE REVIEW

File: tensorboard/plugins/pr_curve/summary.py [hunk 1]
```diff
@@ -148,23 +148,141 @@ def op(
...
+    A summary operation for use in a TensorFlow graph. See docs for the `op`
+    method for details on the float32 tensor produced by this summary.
+  """
+  with tf.name_scope(tag, values=[
+      true_positive_counts,
+      false_positive_counts,
+      true_negative_counts,
+      false_negative_counts,
+      precision,
+      recall
```
Comment (wchargin) [inline tensorboard/plugins/pr_curve/summary.py]: nit: trailing comma

File: tensorboard/plugins/pr_curve/summary.py [hunk 2]
```diff
@@ -148,23 +148,141 @@ def op(
...
+    num_thresholds=None,
+    display_name=None,
+    description=None,
+    collections=None):
+  """A private helper method for generating a tensor summary.
+
+  We use a helper method instead of having `op` directly call `raw_metrics_op`
+  to prevent the scope of `raw_metrics_op` from being embedded within `op`.
+
+  Args:
```
Comment (wchargin) [inline tensorboard/plugins/pr_curve/summary.py]: If the linter doesn't complain, would you mind replacing this whole block with "Arguments are the same as for `raw_metrics_op`"?

File: tensorboard/plugins/pr_curve/summary.py (hunk 1 above)
Comment (chihuahua) [reply tensorboard/plugins/pr_curve/summary.py]: Done.

File: tensorboard/plugins/pr_curve/summary.py (hunk 2 above)
Comment (chihuahua) [reply tensorboard/plugins/pr_curve/summary.py]: Done.

**Pull request merged**